src/us/kbase/workspace/test/database/provenance/ProvenanceTestCommon.java

```diff
@@ -59,4 +62,99 @@ public class ProvenanceTestCommon {
 		validPids.put(" https://this.is.the.url/ ", "https://this.is.the.url/");
 		VALID_PID_MAP = Collections.unmodifiableMap(validPids);
 	}
+
+	public static final List<String> VALID_URL_LIST = Collections.unmodifiableList(Arrays.asList(
+		"http://foo.com/blah_blah",
+		"http://foo.com/blah_blah/",
+		"http://foo.com/blah_blah_(wikipedia)",
+		"http://foo.com/blah_blah_(wikipedia)_(again)",
+		"http://www.example.com/wpstyle/?p=364",
+		"https://www.example.com/foo/?bar=baz&inga=42&quux",
+		"http://✪df.ws/123",
+		"http://userid:password@example.com:8080",
+		"http://userid:password@example.com:8080/",
+		"http://userid@example.com",
+		"http://userid@example.com/",
+		"http://userid@example.com:8080",
+		"http://userid@example.com:8080/",
+		"http://userid:password@example.com",
+		"http://userid:password@example.com/",
+		"http://➡.ws/䨹",
+		"http://⌘.ws",
+		"http://⌘.ws/",
+		"http://foo.com/blah_(wikipedia)#cite-1",
+		"http://foo.com/blah_(wikipedia)_blah#cite-1",
+		"http://foo.com/unicode_(✪)_in_parens",
+		"http://foo.com/(something)?after=parens",
+		"http://☺.damowmow.com/",
+		"http://code.google.com/events/#&product=browser",
+		"http://j.mp",
+		"ftp://foo.bar/baz",
+		"http://foo.bar/?q=Test%20URL-encoded%20stuff",
+		"http://مثال.إختبار",
+		"http://例子.测试",
+		"http://-.~_!$&'()*+,;=:%40:80%2f::::::@example.com",
+		"http://1337.net",
+		"http://a.b-c.de",
+		"http://223.255.255.254"
+	));
+
+	public static final Map<String, String> INVALID_URL_MAP;
+	static {
+		Map<String, String> invalidUrlMap = new HashMap<>();
```

Comment:
TODO: sort this set of dodgy URLs out and winnow it down to create a reasonable test set.

Unfortunately the source of these sample invalid URLs used a slightly different test for URL validity to that used by java, so some of these supposedly invalid URLs are considered OK by java.

There is also an issue with incomplete URLs of the form `http://<special characters here>` where java removes the `//` from the URL string, which makes it more difficult to match up error strings.

Reply:
DONE!

(Decided not to fix the underlying issues with the Java URL library... selfish, I know)